### ISSUE TITLE
Unable to compile from source or install package

### ISSUE THREAD

**Issue by D-Nice** [BitProfile/Xeth#36]
Running Lubuntu 16.04

The Ubuntu 16 package seems to have dependencies exclusive to 15.10 and 16.10 repositories (libjsconcpp0v5 and libcrypto++6 respectively). Once I did get this I get the following when trying to start Xeth:

> ./xeth: symbol lookup error: ./xeth: undefined symbol: _ZTIN8CryptoPP4SHA3E

The Mint 18 package installed without any issues, the Xeth UI pops up for about a minute, and then crashes, with the following log:

> starting qt strategy
> checking ipfs...
> testing ipfs:  "/opt/xeth/vendor/bin/ipfs"  args= ("swarm", "peers")
> "Error: api not running\n"
> initializing ipfs ...
> registering template  "receive"  :  QVariant(bool, true)
> registering template  "profile_item"  :  QVariant(bool, true)
> registering template  "main_page"  :  QVariant(bool, true)
> registering template  "bitprofile"  :  QVariant(bool, true)
> registering template  "transactions"  :  QVariant(bool, true)
> registering template  "transaction_item"  :  QVariant(bool, true)
> registering template  "import_bitprofile"  :  QVariant(bool, true)
> registering template  "generate"  :  QVariant(bool, true)
> registering template  "addressbook"  :  QVariant(bool, true)
> registering template  "import"  :  QVariant(bool, true)
> registering template  "export"  :  QVariant(bool, true)
> registering template  "account_item"  :  QVariant(bool, true)
> registering template  "bitprofile_form"  :  QVariant(bool, true)
> registering template  "accounts"  :  QVariant(bool, true)
> registering template  "account_null"  :  QVariant(bool, true)
> registering template  "settings"  :  QVariant(bool, true)
> registering template  "send"  :  QVariant(bool, true)
> registering template  "change_password"  :  QVariant(bool, true)
> registering template  "contact_item"  :  QVariant(bool, true)
> registering template  "view_bitprofile"  :  QVariant(bool, true)
> starting daemon...
> QProcess: Destroyed while process ("/opt/xeth/vendor/bin/ipfs") is still running.
> checking geth...
> failed to connect, forking new process
> failed to connect, retrying in 1 sec
> failed to connect, retrying in 1 sec
> failed to connect, retrying in 1 sec
> failed to connect, retrying in 1 sec
> synchronizing...
> emitting ready signal
> Illegal instruction (core dumped)

Followed the source build guide then, would fail on multiple occasions even though I had the dependencies mentioned. Had to get additional findboost dependencies, and was able to get up to 93% build with the following error:

> [ 93%] Building CXX object CMakeFiles/xeth.dir/html.cxx.o
> make[2]: **\* No rule to make target '/usr/lib/libcryptopp.so', needed by 'xeth'.  Stop.
> CMakeFiles/Makefile2:532: recipe for target 'CMakeFiles/xeth.dir/all' failed
> make[1]: **\* [CMakeFiles/xeth.dir/all] Error 2
> Makefile:127: recipe for target 'all' failed
> make: **\* [all] Error 2

Would love to try this client.

**Comment by devktor** [comment]
Thank you for reporting it.

Can you please check if libcryptopp-dev is installed? 

```
sudo apt-get install libcryptopp-dev
```

**Comment by devktor** [comment]
Checking Mint crash issue... 

Can you tell me what Desktop are you using: "Mate", "Cinnamon" or "XFCE" ?

On what processor architecture: 64 or 32 Bit?

**Comment by D-Nice** [comment]
libcryptopp-dev is installed, the issue is it was initially only version 5.6.1_9, which is missing SHA-3. That's the most up to date one available in the 16.04 repo. I added 16.10 repo, and upgraded to 5.6.3 but still got the above build error.

Lubuntu 64-bit
Desktop = LXDE

Literally using a fresh install from http://cdimage.ubuntu.com/lubuntu/releases/xenial/release/lubuntu-16.04.1-desktop-amd64.iso

**Comment by devktor** [comment]
I have updated ubuntu build instructions. Please try this : https://github.com/BitProfile/Xeth/blob/master/doc/build_ubuntu14.md

Works for ubuntu (ubuntu, lubuntu etc.) and mint.

Will update deb packages tonight.

**Comment by D-Nice** [comment]
Thanks, installing dependencies now, do note I think you made a typo

> sudo apt-get svn install git cmake make g++ libboost-all-dev libleveldb-dev libjsoncpp-dev libqt5webkit5-dev libappindicator-dev libnotify-dev libgtk2.0-dev

You seem to have put svn before install. Also, I had to actually write subversion instead of svn, this was the command I used: 

> sudo apt-get install subversion git cmake make g++ libboost-all-dev libleveldb-dev libjsoncpp-dev libqt5webkit5-dev libappindicator-dev libnotify-dev libgtk2.0-dev

Will report back if it builds once this part is complete.

**Comment by D-Nice** [comment]
Works now, glad I'm getting to try this great project out now! Absolutely love your UI designs. Would be nice to have a setting, to get some more info on the network synchronization, such as the current block it's on etc... by hovering over the sync bar. And can't wait for this to work with parity!

Closing this as it does build now, keep up the great work.

**Comment by devktor** [comment]
Thanks. Fixed the typo.

Current block, network peers etc. are coming in the next release, which should be quite soon.
